plugins/aws_sts/app/views/samson_aws_sts/_fields.html.erb

```diff
@@ -0,0 +1,17 @@
+<fieldset>
+  <legend>AWS STS Tokens</legend>
+  <div class="col-lg-offset-2">
+    <div class="form-group">
+      <div class="col-lg-6">
+        <p>AWS IAM Role ARN that should be assumed by the stage:</p>
+        <%= form.text_field :aws_sts_iam_role_arn, class: "form-control", placeholder: "e.g. arn:aws:iam::123456789012:role/S3Access" %>
```

Comment:
Does this get validated in any way? How would a user know if the ARN is valid, or whether Samson has permission to assume into that role?

Reply:
I haven't included validation. At run time we get the following exceptions:

`Aws::STS::Client.new.assume_role(role_arn: "1234", role_session_name: "nope")` results in: 
`Aws::STS::Errors::ValidationError: 1 validation error detected: Value '1234' at 'roleArn' failed to satisfy constraint: Member must have length greater than or equal to 20
from /usr/local/bundle/gems/aws-sdk-core-2.10.21/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'`

`Aws::STS::Client.new.assume_role(role_arn: "arn:aws:iam::589470541234:user/samson-sts-test-user", role_session_name: "nope")` results in:
`Aws::STS::Errors::AccessDenied: User arn:aws:iam::589470541234:user/some_user is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::589470541234:user/samson-sts-test-user`

I guess we could validate that the ARN "looks like" an ARN, and allow the deploy to fail and surface the exception if there is a permissions error

Reply:
you can also do `pattern: /\Aarn:aws:iam::\d+:role\/.+\Z/` here and not bother with the model validation :)